### DISCUSSION TITLE
Exposing the `/admin` path on a per-need basis

### DISCUSSION THREAD

**Discussion by sndmnn** [keycloak/keycloak#34096] [Q&A] [answered]
I'm preparing to deploy a Keycloak server to be the identity manager for a couple of applications I'm writing. Everything is going to be hosted on the same machine, so I'm configuring it all behind a reverse proxy. I've been playing with a staging instance before I actually deploy everything.

I've also been exploring the docs for topics relevant to "deploying Keycloak" and I got to the [Exposed path recommendations](https://www.keycloak.org/server/reverseproxy#_exposed_path_recommendations) part. It makes sense, you don't want to expose things that don't need to be exposed for your applications to run. But that way, every time I need to access the admin console, I'd need to SSH into the server and modify the reverse proxy configuration to expose the `/admin` path. And after I'm done, I'd need to obscure it again.

This seems a bit rudimentary and I'm wondering if there's a better way to go about it. I don't have a problem doing it since I'm not going to access the admin console a whole lot after the initial configuration, but I wanted to know if there are other recommendations out there.

**Comment by codespearhead** [comment, marked as answer]
Restrict that endpoint to localhost in the reverse proxy and use a VPN or SSH Tunneling to access it.

**Reply by darius-m** [reply]
Using the `-D` SSH option for a SOCKS proxy on a bastion system, in combination with a browser that allows you to configure a SOCKSv5 proxy (e.g. Firefox) is also a very easy approach.
